### PR TITLE
chore: Remove some logs and errors from console - 1/n

### PR DESCRIPTION
**Pull request by dvj1988**
## Description
This PR removes some of the errors and logs from the console. The following fixes were added to this PR
- Remove the error => Refused to set unsafe header "Expect"
- Remove the aria label warnings
- Remove the logs from service worker in dev mode

Fixes #`Issue Number`  
_or_  
Fixes `Issue URL`
> [!WARNING]  
> _If no issue exists, please create an issue first, and check with the maintainers if the issue is valid._

## Automation

/ok-to-test tags="@tag.All"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/10438062046>
> Commit: 63fbfb22f40508ab298c765c4edbba7c42a125e1
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=10438062046&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.All`
> Spec:
> <hr>Sun, 18 Aug 2024 06:12:03 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **Accessibility Improvements**
	- Added `aria-label` attributes to the `StyledSearchInput` and `Label` components for enhanced screen reader support.
	- Improved the semantic structure of the `Label` component by providing additional context for assistive technologies.

- **Performance Enhancements**
	- Removed the `Expect` header from API requests to streamline the request process.

- **Logging Adjustments**
	- Disabled development logs in the service worker to enhance performance in production environments.

### CODE REVIEW

File: app/client/packages/design-system/widgets/src/components/Label/src/Label.tsx [hunk 1]
```diff
@@ -13,6 +13,8 @@ export const Label = (props: LabelProps) => {
 
   return (
     <HeadlessLabel
+      aria-label={text}
+      aria-labelledby="label"
```
Comment (znamenskii-ilia) [inline app/client/packages/design-system/widgets/src/components/Label/src/Label.tsx:17]: I don't see the element with id `label`. How this should work?

File: app/client/src/serviceWorker.ts [hunk 2]
```diff
@@ -40,7 +40,7 @@ const regexMap = {
 const toPrecache = (self as any).__WB_MANIFEST;
 precacheAndRoute(toPrecache);
 
-self.__WB_DISABLE_DEV_LOGS = false;
+self.__WB_DISABLE_DEV_LOGS = true;
```
Comment (znamenskii-ilia) [inline app/client/src/serviceWorker.ts:43]: 🔥

File: app/client/packages/design-system/widgets/src/components/Label/src/Label.tsx (hunk 1 above)
Comment (dvj1988) [reply app/client/packages/design-system/widgets/src/components/Label/src/Label.tsx:17]: My bad. I am removing this as we aren't passing any ids. The warning should be suppressed if one of them is present.

File: app/client/src/serviceWorker.ts (hunk 2 above)
Comment (KelvinOm) [reply app/client/src/serviceWorker.ts:43]: @znamenskii-ilia @dvj1988 Maybe it's worth considering disabling this for dev mode only? What do you think?

**Review by KelvinOm — APPROVED** [review]

File: app/client/src/serviceWorker.ts (hunk 2 above)
Comment (znamenskii-ilia) [reply app/client/src/serviceWorker.ts:43]: Am I right that you suggest we enable dev logs in prod?

**Review by znamenskii-ilia — APPROVED** [review]

File: app/client/src/serviceWorker.ts (hunk 2 above)
Comment (dvj1988) [reply app/client/src/serviceWorker.ts:43]: The performance team is owning the service worker implementation and we don't need the logs for this on prod/dev at the moment. It makes the console very noisy.
@KelvinOm Are you suggesting we should tie it with the debug LogLevel?

Will merge the PR after we resolve this comment.

**Comment by dvj1988** [comment]
This PR has some fixes for https://github.com/appsmithorg/appsmith-ee/issues/4748

File: app/client/src/serviceWorker.ts (hunk 2 above)
Comment (KelvinOm) [reply app/client/src/serviceWorker.ts:43]: @dvj1988 If we're sure we don't need it, then I'm okay with it.

**Pull request merged**